### PR TITLE
Clean-up and documentation for backend

### PR DESCRIPTION
**Pull request by mgbckr**
Here is a first attempt of documenting the backend functionality. I also moved the corresponding test file to examples since that is more appropriate. Related to #311 and #312 .

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -407,6 +407,25 @@ file ~/.ipython/ipy_user_conf.py to add the following lines::
...
+If no specific backend is specified the default is to use "psutil" which measures RSS aka “Resident Set Size”. 
+In some cases (particularly when tracking child processes) RSS may overestimate memory usage (see `example/example_psutil_memory_full_info.py` for an example).
+For more information on "psutil_pss" (measuring PSS) and "psutil_uss" please refer to:
+https://psutil.readthedocs.io/en/latest/index.html?highlight=memory_info#psutil.Process.memory_full_info 
+
+Currently, the backend can be set via the CLI
+
+    $ python -m memory_profiler --backend psutil my_script.py
+
+and is explosed by the API
```
Comment (fabianp) [inline README.rst:423]:
```suggestion
and is exposed by the API
```

**Review by fabianp — CHANGES_REQUESTED** [review]

**Comment by fabianp** [comment]
thanks! other than the minor typo it looks good to me!

**Comment by fabianp** [comment]
@mgbckr : please feel free to add your name to the end of the README

**Comment by mgbckr** [comment]
Alright, fixed the typo and added my name, thank you :)

**Comment by fabianp** [comment]
great, merged

**Pull request merged**